crates/ruff/src/checkers/imports.rs

```diff
@@ -54,5 +55,45 @@ pub fn check_imports(
             &blocks, python_ast, locator, stylist, settings, autofix,
         ));
     }
-    diagnostics
+    let mut imports: FxHashMap<PathBuf, Vec<Import>> = FxHashMap::default();
+    let mut imports_vec = vec![];
+    for &block in &blocks {
+        block.imports.iter().for_each(|&stmt| match &stmt.node {
+            StmtKind::Import { names } => {
+                // from testing, seems this should only have one entry
+                imports_vec.push(Import {
+                    name: names[0].node.name.clone(),
+                    location: stmt.location,
+                    end_location: stmt.end_location.unwrap(),
+                });
+            }
+            StmtKind::ImportFrom { module, names, .. } => imports_vec.extend(
+                names
+                    .iter()
+                    .map(|name| Import {
+                        name: format!(
+                            "{}.{}",
+                            {
+                                if let Some(n) = module {
+                                    n
+                                } else {
+                                    ""
+                                }
+                            },
+                            name.node.name
+                        ),
+                        location: name.location,
+                        end_location: name.end_location.unwrap(),
+                    })
+                    .collect::<Vec<Import>>(),
+            ),
+            _ => unreachable!("Should only have import statements"),
```

Comment:
Nit: Can we add a small comment explaining what guarantees that there should only be import statements?